Replace Buffer constructor with buffer-from

jasnell opened the pull request:
Adds a dependency on buffer-from

jasnell committed:
Replace Buffer constructor with buffer-from

Adds a dependency on buffer-from

mafintosh commented on index.js:
```diff
@@ -22,7 +23,7 @@ var peek = function(opts, onpeek) {
   var peeker = through.obj({highWaterMark:1}, function(data, enc, cb) {
     if (isObject(data)) return ready(data, null, cb)
-    if (!Buffer.isBuffer(data)) data = new Buffer(data)
+    if (!Buffer.isBuffer(data)) data = Buffer.from(data)
```
i think you meant bufferFrom here

jasnell replied:
Ha! Muscle memory ftw

jasnell commented:
Fixed! Also, my global config setting for `git config core.fileMode false` never seems to want to work and I consistently forget to set it in advance so sometimes a file mode change will sneak through. I try to catch them myself, but they sneak through and the original commit here had it. I fixed that too tho. should be good to go.

mafintosh commented:
1.1.3 :tada:
